### PR TITLE
Feat: BL-15791 update endpoints to match production

### PR DESCRIPTION
**Pull request by JustinJohnston-alt**
## Description

removed check for DVSA reference number
removed extra logging
updated postman collection to remove test for invalid DVSA reference number rejection.

Related issue: [BL-15791](https://dvsa.atlassian.net/browse/BL-15791)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] Feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -61,7 +63,7 @@ An API Usage Key will be provided to you when you register for the service, this
   "recallCampaignStartDate": "2022-01-31"
 }
 ```
-### Update a recall
+### Update a recall as Fixed
```
Comment (mateuszc-kainos) [inline README.md]: Can we clear this up a little bit? Perhaps "Update a recall - set status to Fixed"

File: README.md [hunk 2]
```diff
@@ -76,6 +78,21 @@ An API Usage Key will be provided to you when you register for the service, this
   "repairStatus": "FIXED"
 }
 ```
+
+### Update a recall as Not Fixed
```
Comment (mateuszc-kainos) [inline README.md]: See above

**Review by mateuszc-kainos — APPROVED** [review]

File: src/handler/put.ts [hunk 3]
```diff
@@ -33,33 +33,54 @@ export const handler = async (event: APIGatewayProxyEvent): Promise<APIGatewayPr
...
+  }
+
+  if (vehicleFound.repairStatus === RepairStatus.NOT_FIXED) {
+    if (!allRequiredFieldsUpdateNonfixedRecall(recallUpdate)) {
+      return HttpResponse(StatusCodes.BAD_REQUEST, ExternalApiErrorMessages.InvalidRequestBody);
     }
-    if (!validDateFormat(recallUpdate.rectificationDate)) {
-      return HttpResponse(StatusCodes.BAD_REQUEST, ExternalApiErrorMessages.InvalidDateFormat);
+    if (recallUpdate.repairStatus === RepairStatus.FIXED) {
+      if (vehicleFound) {
```
Comment (UndermountainK) [inline src/handler/put.ts]: Minor: Logic here is a bit odd to me. First we check `vehicleFound.repairStatus` and later we check `vehicleFound`. In my opinion this line is not needed.

File: src/handler/put.ts [hunk 4]
```diff
@@ -33,33 +33,54 @@ export const handler = async (event: APIGatewayProxyEvent): Promise<APIGatewayPr
...
+      }
+    } else {
+      return HttpResponse(StatusCodes.BAD_REQUEST, ExternalApiErrorMessages.VehicleRecallAlreadyNotFixed);
+    }
+  } else {
+    if (!allRequiredFieldsUpdateFixedRecall(recallUpdate)) {
+      return HttpResponse(StatusCodes.BAD_REQUEST, ExternalApiErrorMessages.InvalidRequestBody);
+    }
+    if (recallUpdate.repairStatus === RepairStatus.NOT_FIXED) {
+      if (vehicleFound) {
```
Comment (UndermountainK) [inline src/handler/put.ts]: Minor: same as above

**Review by UndermountainK — APPROVED** [review]
Left some comments but LGTM

File: README.md (hunk 1 above)
Comment (JustinJohnston-alt) [reply README.md]: done

File: README.md (hunk 2 above)
Comment (JustinJohnston-alt) [reply README.md]: done

File: src/handler/put.ts (hunk 3 above)
Comment (JustinJohnston-alt) [reply src/handler/put.ts]: changed to remove extra check

File: src/handler/put.ts (hunk 4 above)
Comment (JustinJohnston-alt) [reply src/handler/put.ts]: changed to remove extra check

File: src/util/validatorsRecall.ts [hunk 5]
```diff
@@ -24,12 +24,24 @@ export const allRequiredFieldsCreateRecall = (recall:RecallsCreateRequest):boole
...
+  const areAllRequiredFieldsPresent:boolean = (
+    checkProperty(recall.repairStatus)
+  );
+  return areAllRequiredFieldsPresent;
+};
+
+export const rectificationDateIsInvalid = (recallClosedDate:string, recallOpenDate:string ):boolean => {
+  const openDate = new Date(recallOpenDate);
+  const closedDate = new Date(recallClosedDate);
+  return closedDate < openDate;
```
Comment (alexisc-kainos) [inline src/util/validatorsRecall.ts:45]: Rectification date can be the same as start date (not just before)

**Review by alexisc-kainos — CHANGES_REQUESTED** [review]

**Review by alexisc-kainos — APPROVED** [review]

**Pull request merged**